intlist.c

```diff
@@ -0,0 +1,269 @@
+      if (((*head)->link) == NULL) {
+         printf ("%d is removed from the list\n", temp->value);
+         free (temp);
+         *head = NULL;
+         return;
+      }
+    //To remove the first element when it's not the only element in the list
+      while ((temp->link) != NULL) {
+         prev = temp;
+         temp = temp->link;
+      }
+      printf ("\n%d is removed from the list\n", temp->value);
+      prev->link = NULL;
+      free (temp);
+      PrintFunc (*head);
+      return;
+   }
+   //To remove the last element of the list 
+   if (n == (lengthOfList - 1)) {
+      struct node* temp;
+      temp = *head;
+      printf ("\n%d is removed from the list\n", temp->value);
+      *head = (*head)->link;
+      PrintFunc (*head);
+      free (temp);
+      temp = NULL;
+      return;
+   }
+   struct node* prev = *head;
+   struct node* temp = *head;
+   while ((temp != NULL) && (count < (cmp))) {
+      prev = temp;
+      temp = temp->link;
+      count++;
+
+   };
+   //tempNext points to the next node from the element to be removed 
+   struct node* tempNext = NULL;
+   if (n == 0) {
+      tempNext = NULL;
+   } else {
+      tempNext = temp->link;
+   }
+   prev->link = tempNext;
+   printf ("\n%d is removed from the list\n", temp->value);
+   free (temp);
+   PrintFunc (*head);
+}
+
+//Gets the element from the given index position
+void Get (struct node** head, int n) {
+   int lengthOfTheList = CountNum (head);
+   int count = 0;
+   int cmp = lengthOfTheList - n - 1;
+   struct node* temp = *head;
+   if (n < 0) {
+      printf ("\nInvalid index , did you mean n = 0? \n");
+      return;
+   }
+   if (n > (lengthOfTheList - 1)) {
+      printf ("\nExceeded length of the list \n");
+      return;
+   }
+   while ((count < cmp) && (temp != NULL)) {
+      temp = temp->link;
+      count++;
+   }
+   printf ("\nThe element at index position %d is %d \n", n, temp->value);
+}
+
+//Removes the first occurence of an element from the list
+void Remove (struct node** head, int n) {
+   int count = (CountNum (head)) - 1;
+   int indexPosition = -1;
+   if (*head == NULL) {
+      printf ("\nThere are no elements to remove \n");
+      return;
+   } else {
+      struct node* temp;
+      temp = *head;
+      while (temp != NULL) {
+         if ((temp->value) == n) {
+            indexPosition = count;
+         }
+         temp = temp->link;
+         count--;
+      }
+      if (indexPosition != -1) {
+         printf ("\nThe earliest occurence of %d is in index position %d \n", n, indexPosition);
+         RemoveAt (head, indexPosition);
+         return;
+      } else {
+         printf ("The given element does not exist in the list \n");
+         return;
+      }
+   }
+}
+
+//Inserts element at the given index position
+void Insert (struct node** head, int n, int pos) {
+   int count = CountNum (head);
+   if ((pos > count) || (pos < 0)) {
+      printf ("\nInvalid insertion \n");
+      return;

```

Comment:
How is the calling function supposed to know that the "insert" failed?

Reply:
I have included a function called CountNum within intlist.c which returns the length of the list. Insert() determines if the index position is valid based on the length of the linked list. Please let me know if my approach could be improved. Thank you.